### PR TITLE
chore(ts): Convert marked

### PR DESCRIPTION
**Pull request by evanpurkhiser**
Tests all pass. Haven't tried to fix the deprecated sanitize warnings yet

### CODE REVIEW

File: src/sentry/static/sentry/app/utils/marked.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+// Only https and mailto, (e.g. no javascript, vbscript, data protocols)
+const safeLinkPattern = /^(https?:|mailto:)/i;
+
+const safeImagePattern = /^https?:\/\/./i;
+
+function isSafeHref(href: string, pattern: RegExp) {
+  try {
+    return pattern.test(decodeURIComponent(unescape(href)));
+  } catch (e) {
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/utils/marked.tsx]:
```suggestion
  } catch {
```

File: src/sentry/static/sentry/app/utils/marked.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+// malicious uris.
+
+class SafeRenderer extends marked.Renderer {
+  link(href: string, title: string, text: string) {
+    // For a bad link, just return the plain text href
+    if (!isSafeHref(href, safeLinkPattern)) {
+      return href;
+    }
+
+    let out = '<a href="' + href + '"';
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/utils/marked.tsx]: Wouldn't it be better to use template literals here?

Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/utils/marked.tsx]: Yeah good call

File: src/sentry/static/sentry/app/utils/marked.tsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    // For a bad image, return an empty string
+    if (this.options.sanitize && !isSafeHref(href, safeImagePattern)) {
+      return '';
+    }
+
+    let out = '<img src="' + href + '" alt="' + text + '"';
+    if (title) {
+      out += ' title="' + title + '"';
+    }
+    out += this.options.xhtml ? '/>' : '>';
```
Comment (markstory) [inline src/sentry/static/sentry/app/utils/marked.tsx]: Another place you could use template literals. Also our application is not xhtml so you could remove the related option.

**Review by markstory — APPROVED** [review]

**Pull request merged**